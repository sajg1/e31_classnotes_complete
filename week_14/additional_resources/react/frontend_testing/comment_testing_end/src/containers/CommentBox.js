import React, {Component} from "react";
import CommentList from "../components/CommentList";
import CommentForm from "../components/CommentForm";

class CommentBox extends Component {

  constructor(props) {
    super(props);
    this.state = {
      data: [{
          id: 1,
          author: "Colin",
          text: "Our comments are still hanging around."
        },
        {
          id: 2,
          author: "Jarrod",
          text: "React has converted me, I was wrong to be #teamjava."
        }
      ]
    };
    this.handleCommentSubmit = this.handleCommentSubmit.bind(this);
    this.handleCommentDelete = this.handleCommentDelete.bind(this);

  }

  handleCommentSubmit(submittedComment) {
    submittedComment.id = Date.now();
    const updatedComments = [...this.state.data, submittedComment];
    this.setState({
      data: updatedComments
    });
  }

  handleCommentDelete(evt) {
    evt.preventDefault();
    this.setState({
      data: []
    });
  }


  render() {
    return ( 
      <div className = "comment-box" >
        <h2>Add a Comment</h2> 
        <CommentForm onCommentSubmit = {this.handleCommentSubmit}/> 
        <h2>Comments</h2> 
        <CommentList data = {this.state.data}/> 
        <form onSubmit = {this.handleCommentDelete}>
          <input 
            id = "delete-button"
            type = "submit"
            value = "Delete Comments" 
          />
        </form> 
      </div>
    );
  }
}

export default CommentBox;
