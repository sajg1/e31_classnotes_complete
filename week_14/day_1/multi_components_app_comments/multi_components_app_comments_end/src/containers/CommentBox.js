import React, { Component } from "react";
import CommentList from "../components/CommentList";

class CommentBox extends Component {

  constructor(props) {
    super(props);
    this.state = {
      data: [
        { id: 1,
          author: "Brendan Eich",
          text: "Always bet on JavaScript."
        },
        { id: 2,
          author: "Reg Braithwaite",
          text: "The strength of JavaScript is that you can do anything. The weakness is that you will."
        }
      ]
    };
  }

  render() {
    return (
      <div className="comment-box">
        <h2>Comments</h2>
        <CommentList data={this.state.data} />
      </div>
    );
  }
}

export default CommentBox;
