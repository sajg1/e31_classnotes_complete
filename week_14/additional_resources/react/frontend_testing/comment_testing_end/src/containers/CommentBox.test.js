import React from "react";
import CommentBox from "./CommentBox";
import CommentList from "../components/CommentList";
import CommentForm from "../components/CommentForm";
import { shallow, mount } from "enzyme";

describe("CommentBox", () => {

  let wrapper;

  beforeEach(() => wrapper = shallow(<CommentBox />));

  it("should render correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should render a <div>", () => {
    expect(wrapper.find("div").length).toEqual(1);
  });

  it("should render two <h2>s", () => {
    expect(wrapper.find("h2").length).toEqual(2);
  });

  it("should render a CommentForm", () => {
    expect(wrapper.containsMatchingElement(<CommentForm />)).toEqual(true);
  });

  it("should render a CommentList", () => {
    expect(wrapper.containsMatchingElement(<CommentList />)).toEqual(true);
  });

  it("should render CommentForm and CommentList together", () => {
    expect(wrapper.containsAllMatchingElements([
      <CommentForm />,
      <CommentList />
    ])).toEqual(true);
  });


});


describe("Mounted CommentBox", () => {

  it("should call handleCommentDelete when the delete button is clicked", () => {
    const spy = jest.spyOn(CommentBox.prototype, "handleCommentDelete");
    const wrapper = mount(<CommentBox/>);
    wrapper.find("#delete-button").simulate("submit");
    expect(spy).toHaveBeenCalledTimes(1);
  });

  it("should call handleCommentSubmit when a comment is submitted", () => {
    const spy = jest.spyOn(CommentBox.prototype, "handleCommentSubmit");
    const wrapper = mount( < CommentBox /> );
    const form = wrapper.find("CommentForm");
    form.instance().setState({author: "Juan", text: "This is going to be really hard."});
    form.find("#submit-button").simulate("submit");
    expect(spy).toHaveBeenCalledTimes(1);
  });

});