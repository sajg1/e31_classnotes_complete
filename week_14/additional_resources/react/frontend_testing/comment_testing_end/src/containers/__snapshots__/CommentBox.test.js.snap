// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentBox should render correctly 1`] = `
<div
  className="comment-box"
>
  <h2>
    Add a Comment
  </h2>
  <CommentForm
    onCommentSubmit={[Function]}
  />
  <h2>
    Comments
  </h2>
  <CommentList
    data={
      Array [
        Object {
          "author": "Colin",
          "id": 1,
          "text": "Our comments are still hanging around.",
        },
        Object {
          "author": "Jarrod",
          "id": 2,
          "text": "React has converted me, I was wrong to be #teamjava.",
        },
      ]
    }
  />
  <form
    onSubmit={[Function]}
  >
    <input
      id="delete-button"
      type="submit"
      value="Delete Comments"
    />
  </form>
</div>
`;
